### PR TITLE
Corrected physics.vector printing dots on top for ASCII to indicate derivatives to Apostrophe

### LINKED ISSUE #15884: ASCII printer generates Unicode (at least in physics.vector)
As noted in https://github.com/sympy/sympy/pull/15849 the printing function in physics.vector adds dots on top of variables to indicate derivatives (Newton style). However, it seems like these are not properly converted to ASCII later when the ASCII version is derived. This should be sorted out in pretty_symbology and replace the dots with a proper number of `'`.

Note that there are currently tests that will have to be changed, e.g. https://github.com/sympy/sympy/blob/cb8a524e930266fe2f4cda39d7d0b1523c4fa38b/sympy/physics/vector/tests/test_printing.py#L239

where the expected result should be `'omega\'\'\' n_x'` (or `"omega''' n_x"` which may be easier to read and type...).

### PR DESCRIPTION
**Pull request by shivanikohlii**
#### References to other Issues or PRs
Fixes #15884

#### Brief description of what is fixed or changed
Fixes the Ascii generates Unicode issue for physics.vector. Now physics.vector derivative displays apostrophes instead of Newton dots. Also modified test_printing.py in physics/vector/tests to check for the right issues. 

#### Release Notes

* physics.vector
  * Added apostrophe on the ASCII printing of derivatives of physics.vector instead of the (Unicode) Newton dots.

### CODE REVIEW

**Comment by shivanikohlii** [comment]
@oscargus hi! could you please help me with why it is not passing the checks? It is my first pull request!

File: sympy/physics/vector/tests/test_printing.py [hunk 1]
```diff
@@ -229,25 +229,29 @@ def test_vector_derivative_printing():
...
 
     assert v._latex() == r'\ddddot{\omega}\mathbf{\hat{n}_x}'
     assert unicode_vpretty(v) == u('ω⃜ n_x')
-    assert ascii_vpretty(v) == u('ome⃜ga n_x')
+    assert ascii_vpretty(v) == u("omega'''' n_x")
+
+    # # Fifth order
+    # v = omega.diff().diff().diff().diff().diff() * N.x
+
+    # assert v._latex() == r'\frac{d^{5}}{d t^{5}} \omega{\left(t \right)}\mathbf{\hat{n}_x}'
```
Comment (oscargus) [inline sympy/physics/vector/tests/test_printing.py]: Please keep these tests. They should still work.

**Comment by oscargus** [comment]
Not sure why the tests are not even running. I fixed the release note though (basically, you needed a space between the * and the text.

The code looks quite good. I think it makes sense to put an upper limit on the number of apostrophes to, say, four, and then rely on the standard printing. See the other code.

**Comment by asmeurer** [comment]
The tests should run eventually. But I can already see from the diff that you have trailing white space that will need to be fixed. You can run `./bin/test quality` to run the code quality checks. 

Also it is not necessary to mention the tests in the release notes. They release notes are for end-users, who do not care about tests changes.

**Comment by asmeurer** [comment]
Perhaps the tests are not running because of the merge conflicts.

**Comment by shivanikohlii** [comment]
@asmeurer I'll fix the merge conflict. The only issue is when we fixed the trailing whitespace issue and the individual test was passing but the test suite was failing still. But I'll work on it today and get back to you guys, thank you for all your help!

**Comment by moorepants** [comment]
The physics vector is currently only designed to handle time as an implicit variable. It is not customary to use a prime to indicate the time derivative. My vote would be to retain the Leibniz notation dx/dt (but maybe have it display in one line instead of three). With the prime there is no knowledge from the reader what the derivative is taken with respect to.

**Comment by oscargus** [comment]
@moorepants while I cannot argue that you are wrong as such, isn't that a general problem using primes? And in physics vector it would be, sort of, more obvious that it is wrt time than in the general case? As a layman (with a PhD in EE though), this would seems like a quite good option, at least compared to using unicode in a non-unicode terminal...

Either way, changing to use Leibniz should be rather straightforward (given that the code is based on the current master as there is a check for higher-than-fourth-order-derivatives, where the _use_unicode check can be easily added, still that would be multi-line Leibniz).

How would a single-line derivative look like? `d(omega)/dt`? `d/dt omega`? Several derivatives?

@shivanikohlii please update to the latest master as there are some related changes in there.

Then, you can change:          https://github.com/sympy/sympy/blob/c267d554e16f0392af2b22a2922cbe0db7e8c798/sympy/physics/vector/printing.py#L189
to `if dot_i >= 5 or not self._use_unicode:` to get the non-unicode, multi-line Leibniz approach.

**Comment by moorepants** [comment]
Maybe we should do `omega''(t)`. If you leave the function argument in place, then the prime works clearly.

**Comment by shivanikohlii** [comment]
@asmeurer When I am running the doc tests locally they're running successfully but failing here. I'm not sure what's happening. 
image

**Comment by oscargus** [comment]
I think you are only testing the file `test_code_quality.py`. Run it as
`./bin/test quality` and see if it differs.

There is clearly an additional empty line at the end of one of the files, so somehow that is not currently tested. My experience is that it covers all files well with the approach above.

(Still I note that the tests take some time, so I understand that you expect it to test everything.)

**Comment by shivanikohlii** [comment]
@oscargus I ran ./bin/test quality and all the tests there pass too. 
 screen shot 2019-02-14 at 8 03 45 am

**Comment by jksuom** [comment]
This test fails:
```
$ bin/test sympy/physics/vector/tests/test_printing.py
============================= test process starts ==============================
executable:         /usr/bin/python  (2.7.6-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 2.0.2
numpy:              1.10.2
random seed:        23595098
hash randomization: on (PYTHONHASHSEED=1911466581)

sympy/physics/vector/tests/test_printing.py[9] ........F                  [FAIL]

________________________________________________________________________________
_ sympy/physics/vector/tests/test_printing.py:test_vector_derivative_printing __
Traceback (most recent call last):
  File "/home/jks/test/sympy/sympy/physics/vector/tests/test_printing.py", line 232, in test_vector_derivative_printing
    assert ascii_vpretty(v) == u("omega'' n_x")
AssertionError

============= tests finished: 8 passed, 1 failed, in 0.66 seconds ==============
DO *NOT* COMMIT!
```

**Comment by shivanikohlii** [comment]
> This test fails:
> 
> ```
> $ bin/test sympy/physics/vector/tests/test_printing.py
> ============================= test process starts ==============================
> executable:         /usr/bin/python  (2.7.6-final-0) [CPython]
> architecture:       64-bit
> cache:              yes
> ground types:       gmpy 2.0.2
> numpy:              1.10.2
> random seed:        23595098
> hash randomization: on (PYTHONHASHSEED=1911466581)
> 
> sympy/physics/vector/tests/test_printing.py[9] ........F                  [FAIL]
> 
> ________________________________________________________________________________
> _ sympy/physics/vector/tests/test_printing.py:test_vector_derivative_printing __
> Traceback (most recent call last):
>   File "/home/jks/test/sympy/sympy/physics/vector/tests/test_printing.py", line 232, in test_vector_derivative_printing
>     assert ascii_vpretty(v) == u("omega'' n_x")
> AssertionError
> 
> ============= tests finished: 8 passed, 1 failed, in 0.66 seconds ==============
> DO *NOT* COMMIT!
> ```

It's passing in the dev environment, am I missing something? @jksuom 
image

**Comment by shivanikohlii** [comment]
> This test fails:
> 
> ```
> $ bin/test sympy/physics/vector/tests/test_printing.py
> ============================= test process starts ==============================
> executable:         /usr/bin/python  (2.7.6-final-0) [CPython]
> architecture:       64-bit
> cache:              yes
> ground types:       gmpy 2.0.2
> numpy:              1.10.2
> random seed:        23595098
> hash randomization: on (PYTHONHASHSEED=1911466581)
> 
> sympy/physics/vector/tests/test_printing.py[9] ........F                  [FAIL]
> 
> ________________________________________________________________________________
> _ sympy/physics/vector/tests/test_printing.py:test_vector_derivative_printing __
> Traceback (most recent call last):
>   File "/home/jks/test/sympy/sympy/physics/vector/tests/test_printing.py", line 232, in test_vector_derivative_printing
>     assert ascii_vpretty(v) == u("omega'' n_x")
> AssertionError
> 
> ============= tests finished: 8 passed, 1 failed, in 0.66 seconds ==============
> DO *NOT* COMMIT!
> ```

Am I missing something @jksuom? 
image

**Comment by shivanikohlii** [comment]
@oscargus Hey! we changed it to the Leibniz format as discussed above. Should I change the pull request name? Thanks!

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            "type": "python",
+            "request": "attach",
+            "localRoot": "${workspaceRoot}",
+            "remoteRoot": "${workspaceRoot}",
+            "port": 3000,
+            "secret": "my_secret",
+            "host": "localhost"
+        }
+    ]
+}
```
Comment (sylee957) [inline .vscode/launch.json]: I think this file should be deleted

Comment (shivanikohlii) [reply .vscode/launch.json]: Thank you, done it.

**Review by sylee957 — COMMENTED** [review]
I think some changes in #15891 are being reverted here.
Are they relevant?

**Comment by oscargus** [comment]
Yes, they should be kept. Problem is, I guess, that those changes happened while this was developed, so to solve the merge conflicts simply the old version was used rather than the updated one.

I think that ideally one should use the apostrophe version, but add `(t)` as suggested by @moorepants  in https://github.com/sympy/sympy/pull/15962#issuecomment-462875528 The current approach solves the Unicode problem, but still prints in a much more verbose way than was originally thought.

**Comment by moorepants** [comment]
> I think that ideally one should use the apostrophe version, but add `(t)` as suggested by @moorepants

I agree with this.

**Comment by oscargus** [comment]
Looks good!

If you can please remove the no longer required (and therefore unused) code, this should be good to go!

**Comment by oscarbenjamin** [comment]
Does this affect #14492?

**Comment by oscargus** [comment]
Good question. The things is that Sympy probably prints the dot but the terminal doesn't render it correctly, there are several instances of that. I can try to look more carefully at #14492 as I haven't noticed that.

There is also a question if the dot is on the actual symbol or the subscript... For the current PR it should only improve the situation though.

**Comment by shivanikohlii** [comment]
@oscargus the unused code has been removed. Have a look.

**Comment by sylee957** [comment]
I see a new entry had been added to gitignore.
Though I don’t have any strong opinion about this, since I see SymPy project already contains some loose editor-specific configuration files and blacklists.

**Comment by shivanikohlii** [comment]
@sylee957 I can remove the gitignore file as it is not relevant to this PR but I think in general it might be useful to have .vscodefiles in the gitignore.

**Comment by oscargus** [comment]
I think it looks good! I noticed that first-order printing is not tested for the pretty printer (as the comment says, it is only for LaTeX). If you can add that as well it would be great. But I think the code as such is OK from what I can tell. Thanks!

**Comment by oscargus** [comment]
Looks good! Thanks!

**Pull request merged**